crates/uv/src/commands/project/run.rs

```diff
@@ -563,6 +549,44 @@ pub(crate) async fn run(
                         .flatten();
                 }
             } else {
+                // Validate the requested dependency groups.
+                for group in dev.groups().into_iter().flatten() {
+                    if !project
+                        .pyproject_toml()
+                        .dependency_groups
+                        .as_ref()
+                        .is_some_and(|groups| groups.contains_key(group))
+                    {
+                        return Err(anyhow::anyhow!(
+                            "Group `{group}` is not defined in the project's `dependency-group` table"
+                        ));
+                    }
+                }
+
+                // Determine the default groups to include.
+                let defaults = if let Some(defaults) = project
+                    .pyproject_toml()
+                    .tool
+                    .as_ref()
+                    .and_then(|tool| tool.uv.as_ref().and_then(|uv| uv.default_groups.as_ref()))
+                {
+                    for group in defaults {
+                        if !project
+                            .pyproject_toml()
+                            .dependency_groups
+                            .as_ref()
+                            .is_some_and(|groups| groups.contains_key(group))
+                        {
+                            return Err(anyhow::anyhow!(
+                                "Default group `{group}` is not defined in the project's `dependency-group` table"
+                            ));
+                        }
+                    }
+                    defaults.clone()
+                } else {
+                    vec![DEV_DEPENDENCIES.clone()]
```

Comment:
Can we avoid repeating this block in each command?

Reply:
Done